server/src/main/java/org/elasticsearch/ingest/IngestStats.java

```diff
@@ -242,19 +243,54 @@ IngestStats build() {
     /**
      * Container for pipeline stats.
      */
-    public record PipelineStat(String pipelineId, Stats stats) {
+    public record PipelineStat(String pipelineId, Stats stats, ByteStats byteStats) {
         static List<PipelineStat> merge(List<PipelineStat> first, List<PipelineStat> second) {
-            var totalsPerPipeline = new HashMap<String, Stats>();
+            var totalsPerPipeline = new HashMap<String, PipelineStat>();
 
-            first.forEach(ps -> totalsPerPipeline.merge(ps.pipelineId, ps.stats, Stats::merge));
-            second.forEach(ps -> totalsPerPipeline.merge(ps.pipelineId, ps.stats, Stats::merge));
+            first.forEach(ps -> totalsPerPipeline.merge(ps.pipelineId, ps, PipelineStat::merge));
+            second.forEach(ps -> totalsPerPipeline.merge(ps.pipelineId, ps, PipelineStat::merge));
 
             return totalsPerPipeline.entrySet()
                 .stream()
-                .map(v -> new PipelineStat(v.getKey(), v.getValue()))
+                .map(v -> new PipelineStat(v.getKey(), v.getValue().stats, v.getValue().byteStats))
                 .sorted(PIPELINE_STAT_COMPARATOR)
                 .toList();
         }
+
+        private static PipelineStat merge(PipelineStat first, PipelineStat second) {
+            assert first.pipelineId.equals(second.pipelineId) : "Can only merge stats from the same pipeline";
+            return new PipelineStat(
+                first.pipelineId,
+                Stats.merge(first.stats, second.stats),
+                ByteStats.merge(first.byteStats, second.byteStats)
+            );
+        }
+    }
+
+    /**
+     * Container for ingested byte stats
+     */
+    public record ByteStats(long bytesIngested, long bytesProduced) implements Writeable, ToXContentFragment {
+        public ByteStats(StreamInput in) throws IOException {
+            this(in.readVLong(), in.readVLong());
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeVLong(bytesIngested);
+            out.writeVLong(bytesProduced);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.field("ingested_in_bytes", bytesIngested);
+            builder.field("produced_in_bytes", bytesProduced);
```

Comment:
We should only add these fields to the result for the first pipeline, so we can just emit them if either ingested or produces bytes is non-zero